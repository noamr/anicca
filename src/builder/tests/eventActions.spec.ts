/// </ reference "jest" />
import nearley from 'nearley'
const grammar = require('../eventActions.ne')

const parse = (str: string) => new nearley.Parser(grammar).feed(str).finish()[0]

describe('controllerActions', () => {
    it('actions', () => {
        expect(parse('dispatch someEvent to someController')).toMatchSnapshot()
        expect(parse('run script')).toMatchSnapshot()
    })
})