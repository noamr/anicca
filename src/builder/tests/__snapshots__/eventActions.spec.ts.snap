// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`controllerActions actions 1`] = `
Object {
  "event": "someEvent",
  "target": "someController",
  "type": "Dispatch",
}
`;

exports[`controllerActions actions 2`] = `
Object {
  "type": "RunScript",
}
`;
